### PR TITLE
doc: Update commands and options in radosgw-admin

### PR DESCRIPTION
**Pull request by joscollin**
- Updated the COMMANDS section in RADOSGW-ADMIN doc page, using the latest commands from rgw_admin.cc::usage()
- Updated all OPTIONS sections in RADOSGW-ADMIN doc page, using the latest options from rgw_admin.cc::usage()

Signed-off-by: Jos Collin <jcollin@redhat.com>

### CODE REVIEW

**Comment by joscollin** [comment]
@oritwas Please review. But DNM, as I will do one more commit in this PR, which is in progress. Thanks.

File: doc/man/8/radosgw-admin.rst [hunk 1]
```diff
@@ -89,12 +95,63 @@ which are as follows:
 :command:`bucket check`
   Check bucket index.
 
+:command:`bucket reshard`
+  Reshard a bucket.
+
+:command:`bucket sync enable`
+  Enable bucket sync.
+
+:command:`bucket sync disable`
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: bucket disable should come before bucket enable

File: doc/man/8/radosgw-admin.rst [hunk 2]
```diff
@@ -89,12 +95,63 @@ which are as follows:
...
 :command:`object rm`
   Remove an object.
 
+:command:`object stat`
+  Stat an object for its metadata.
+
 :command:`object unlink`
   Unlink object from bucket index.
 
+:command:`objects expire`
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: Please use the same text as the help command:
run expired objects cleanup

File: doc/man/8/radosgw-admin.rst [hunk 3]
```diff
@@ -104,33 +161,126 @@ which are as follows:
...
+:command:`zonegroup remove`
+  Remove a zone from a zonegroup.
+
+:command:`zonegroup rename`
+  Rename a zone group.
 
-:command:`region list`
-  List all regions set on this cluster.
+:command:`zonegroup list`
+  List all the zone groups set on this cluster.
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: remove the "the"

File: doc/man/8/radosgw-admin.rst [hunk 4]
```diff
@@ -260,9 +459,9 @@ Options
 
    The e-mail address of the user.
 
-.. option:: --display-name=name
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: can you move it after --access as it is in the help command?

File: doc/man/8/radosgw-admin.rst [hunk 5]
```diff
@@ -288,13 +487,30 @@ Options
...
+
 .. option:: --system
 
-	Set the system flag on the user.
+   Set the system flag on the user.
 
 .. option:: --bucket=bucket
 
-   Specify the bucket name.
+   Specify the bucket name. It is also used with quota command.
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: This is used with many other commands (like metadata, reshard) I would just keep the original text

File: doc/man/8/radosgw-admin.rst [hunk 6]
```diff
@@ -323,44 +543,164 @@ Options
...
+
+.. option:: --zonegroup-id=<id>
+
+   The zonegroup id.
+
+.. option:: --zonegroup-new-name=<name>
+
+   The new name of the zonegroup.
+   
 .. option:: --rgw-region=<region>
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: it is a good opportunity to remove this field

File: doc/man/8/radosgw-admin.rst [hunk 7]
```diff
@@ -323,44 +543,164 @@ Options
...
+
+.. option:: --staging
+
+   Get the staging period info.
+
+.. option:: --master
+
+   Set as master.
+
+.. option:: --master-zone=<id>
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: missing --master-url and --master-zonegroup

File: doc/man/8/radosgw-admin.rst [hunk 8]
```diff
@@ -323,44 +543,164 @@ Options
...
+   Set the list of zones to sync from.
+
+.. option:: --sync-from-rm=[zone-name][,...]
+
+   Remove the zones from list of zones to sync from.
 
 .. option:: --fix
 
-	Besides checking bucket index, will also fix it.
+	Besides checking bucket index, fix it too.
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: this sounds wrong, lets keep the old version

File: doc/man/8/radosgw-admin.rst [hunk 9]
```diff
@@ -323,44 +543,164 @@ Options
...
+
+.. option:: --zonegroup-new-name=<name>
+
+   The new name of the zonegroup.
+   
 .. option:: --rgw-region=<region>
 
 	Region in which radosgw is running.
 
-.. option:: --rgw-zone=<zone>
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: this should not be removed

File: doc/man/8/radosgw-admin.rst [hunk 10]
```diff
@@ -323,44 +543,164 @@ Options
...
+
+.. option:: --master
+
+   Set as master.
+
+.. option:: --master-zone=<id>
+
+   Master zone id.
+
+.. option:: --realm-id=<id>
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: missing --rgw-realm and -rgw-zoengroup

**Comment by joscollin** [comment]
@oritwas Thanks for the review. Could you please review this PR first https://github.com/ceph/ceph/pull/18302, so that your comments can be addressed appropriately in the doc ? Thanks.

**Comment by joscollin** [comment]
@oritwas I think we can avoid some of your comments in this PR, if https://github.com/ceph/ceph/pull/18302 is agreed / merged.

**Comment by joscollin** [comment]
@oritwas Fixed your review comments. Please verify. Thanks.

**Comment by joscollin** [comment]
@oritwas Ping

File: doc/man/8/radosgw-admin.rst [hunk 11]
```diff
@@ -424,9 +768,9 @@ Options
 
 	List of caps (e.g., "usage=read, write; user=read".
 
-.. option:: --compression=<compression-algorithm>
+.. option:: --compression=<type>
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: I prefer the original text

File: doc/man/8/radosgw-admin.rst [hunk 12]
```diff
@@ -424,9 +768,9 @@ Options
...
 
-    Placement target compression algorithm (lz4|snappy|zlib|zstd)
+    Placement target compression type (plugin name or empty/none).
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: here too

File: doc/man/8/radosgw-admin.rst [hunk 13]
```diff
@@ -455,20 +814,16 @@ Quota Options
 
 .. option:: --max-size
 
-	Specify max size (in bytes, negative value to disable).
+	Specify max size (negative value to disable).
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: how about: Specify max size in bytes

File: doc/man/8/radosgw-admin.rst [hunk 14]
```diff
@@ -455,20 +814,16 @@ Quota Options
...
 .. option:: --quota-scope
 
-	Scope of quota (bucket, user).
+	The scope of quota (bucket, user).
 
 
 Orphans Search Options
 ======================
 
-.. option:: --pool
```
Comment (oritwas) [inline doc/man/8/radosgw-admin.rst]: This is a option used by orphans find

File: doc/man/8/radosgw-admin.rst (hunk 4 above)
Comment (joscollin) [reply doc/man/8/radosgw-admin.rst]: Fixed

File: doc/man/8/radosgw-admin.rst (hunk 7 above)
Comment (joscollin) [reply doc/man/8/radosgw-admin.rst]: These were already removed in rgw_admin.cc.

File: doc/man/8/radosgw-admin.rst (hunk 11 above)
Comment (joscollin) [reply doc/man/8/radosgw-admin.rst]: I was making it similar to the text in rgw_admin.cc. Anyway, reverted it.

File: doc/man/8/radosgw-admin.rst (hunk 12 above)
Comment (joscollin) [reply doc/man/8/radosgw-admin.rst]: I was making it similar to the text in rgw_admin.cc. Anyway, reverted it.

File: doc/man/8/radosgw-admin.rst (hunk 13 above)
Comment (joscollin) [reply doc/man/8/radosgw-admin.rst]: Please see https://github.com/joscollin/ceph/blob/a12ded9805024d334d9e625a8324463e8f1a4b88/src/rgw/rgw_admin.cc#L308. It is not just `in bytes`, but in other sizes too. That is the reason I removed `in bytes` here.

File: doc/man/8/radosgw-admin.rst (hunk 14 above)
Comment (joscollin) [reply doc/man/8/radosgw-admin.rst]: Please see L535 in the changed file. There were two `--pool`s and I have merged them into one. If you want I can revert. Please suggest.

File: doc/man/8/radosgw-admin.rst (hunk 13 above)
Comment (oritwas) [reply doc/man/8/radosgw-admin.rst]: can you use the same text as in the code?

File: doc/man/8/radosgw-admin.rst (hunk 14 above)
Comment (oritwas) [reply doc/man/8/radosgw-admin.rst]: you can keep it

File: doc/man/8/radosgw-admin.rst (hunk 13 above)
Comment (joscollin) [reply doc/man/8/radosgw-admin.rst]: Done. Please verify.

File: doc/man/8/radosgw-admin.rst (hunk 14 above)
Comment (joscollin) [reply doc/man/8/radosgw-admin.rst]: Done. Please verify.

**Comment by oritwas** [comment]
jenkins test this please

**Comment by joscollin** [comment]
@oritwas Now there is an error, as we have reverted the `--pool` option. Please suggest how to proceed. Thanks.

```
Warning, treated as error:
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/man/8/radosgw-admin.rst:830:Duplicate explicit target name: "cmdoption-radosgw-admin--pool".
Build step 'Execute shell' marked build as failure
```

**Comment by oritwas** [comment]
There was a misunderstanding as I was fine with the --pool change.
And if reverting it causes an error than the change is needed

**Comment by joscollin** [comment]
@oritwas Docs build passed. Please verify and merge.

**Pull request merged**